server/db/migrations/20210504143912_badges_for_awards.js

```diff
@@ -0,0 +1,35 @@
+const { idGenerator } = require('../id_generator');
+
+exports.up = (knex) => {
+  return knex.schema
+    .raw(idGenerator)
+    .createTable('triggers', table => {
+      table.text('id').primary().notNullable().defaultTo(knex.raw('next_id()'));
+      table.text('trigger');
+      table.text('description');
+    })
+    .createTable('badges', table => {
+      table.text('id').primary().notNullable().defaultTo(knex.raw('next_id()'));
+      table.text('creator_id').references('users.id')
+        .onDelete('SET NULL')
+        .onUpdate('CASCADE');
+      table.text('name');
+      table.text('slug');
+      table.text('badge_uri');
+      table.text('trigger').references('triggers.id')
+        .onDelete('SET NULL')
+        .onUpdate('CASCADE');
+      table.integer('points').defaultTo(1);
+      table.text('description');
+      table.dateTime('expiry');
+      table.boolean('is_deleted').defaultTo(false);
```

Comment:
there is a `publish` status (defaults to true)

Reply:
Could you be clear in what you are saying?

Reply:
each badge has a status if it's either published or not. 
 Is that more clear?